### PR TITLE
Upgrade to v2.31.1 of the AzureRM Terraform Provider

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #726
Fixes: #727

### CODE REVIEW

File: provider/resource.go [hunk 1]
```diff
@@ -43,88 +43,89 @@ const (
...
+	azureCore                  = "Core"                  // Base Resources
+	azureCosmosDB              = "CosmosDB"              // Cosmos DB
+	azureCostManagement        = "CostManagement"        // CostManagement
+	azureDashboard             = "Dashboard"             // Dashboard
+	azureDatabaseMigration     = "DatabaseMigration"     // Database Migration
+	azureDataFactory           = "DataFactory"           // Data Factory
+	azureDatalake              = "DataLake"              // Data Lake
+	azureDataShare             = "DataShare"             // DataShare
+	azureDataBricks            = "DataBricks"            // DataBricks
+	azureDesktopVirtualization = "DesktopVirtualization" // DesktopVirtualization
```
Comment (mikhailshilkov) [inline provider/resource.go:73]: I believe the service is "Virtual Desktop" - the same as the TF name. Why did you pick "Desktop Virtualization"?

Comment (stack72) [reply provider/resource.go:73]: I picked it as that was in the TF module - https://github.com/terraform-providers/terraform-provider-azurerm/pull/8605/files#diff-d626fff06c84ff73c96b49db331f82ceR33

Will change it now though :)

Comment (mikhailshilkov) [reply provider/resource.go:73]: I think they are right, the same name is used in Open API specs: https://github.com/Azure/azure-rest-api-specs/tree/master/specification/desktopvirtualization/resource-manager

Let's go ahead with the current version, sorry for the confusion!

Comment (stack72) [reply provider/resource.go]: Ah I forgot to change this back - I will do this now

**Review by mikhailshilkov — APPROVED** [review]

**Pull request merged**